### ISSUE TITLE
no Xauthority is created

### ISSUE THREAD

**Issue by Kingdread** [The-Compiler/pytest-xvfb#5]
Some libraries (namely [python-xlib](https://github.com/python-xlib/python-xlib)) seem to require a valid Xauthority file, either at `$XAUTHORITY` or by default at `~/.Xauthority`, see the (buried) errors in [this run](https://api.travis-ci.org/jobs/152048831/log.txt?deansi=true):

```
2016-08-13 14:07:50,431 qutewm     INFO       qutewm_sub:__init__ initializing
Traceback (most recent call last):
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/xauth.py", line 45, in __init__
    raw = open(filename, 'rb').read()
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/.Xauthority'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/qutebrowser.git/tests/end2end/fixtures/qutewm_sub.py", line 348, in <module>
    main()
  File "/home/user/qutebrowser.git/tests/end2end/fixtures/qutewm_sub.py", line 343, in main
    wm = QuteWM()
  File "/home/user/qutebrowser.git/tests/end2end/fixtures/qutewm_sub.py", line 84, in __init__
    self.dpy = Display()
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/display.py", line 89, in __init__
    self.display = _BaseDisplay(display)
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/display.py", line 71, in __init__
    protocol_display.Display.__init__(self, *args, **keys)
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/protocol/display.py", line 68, in __init__
    name, host, displayno)
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/support/connect.py", line 101, in get_auth
    return mod.get_auth(sock, dname, host, dno)
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/support/unix_connect.py", line 117, in new_get_auth
    au = xauth.Xauthority()
  File "/home/user/qutebrowser.git/.tox/py34/lib/python3.4/site-packages/Xlib/xauth.py", line 47, in __init__
    raise error.XauthError('~/.Xauthority: %s' % err)
Xlib.error.XauthError: ~/.Xauthority: [Errno 2] No such file or directory: '/home/user/.Xauthority'
```

Before writing hacky workarounds in the clients, I think it's best to add this feature to pytest-xvfb itself, maybe behind a configuration option, as generating such a token requires `xauth` to be installed. Using Xvfb directly does not automatically generate such a token, though wrapper scripts like `xvfb-run` seem to do.

I've written a small [patch](https://github.com/The-Compiler/pytest-xvfb/compare/master...Kingdread:master) that adds this functionality, if you're interested I can open a PR.

* [A small how-to with shell commands](http://modb.oce.ulg.ac.be/mediawiki/index.php/Xvfb)
* [xvfb-run source](https://gist.github.com/tyleramos/3744901) (though you can cat this with `cat $(which xvfb-run)`)

**Comment by The-Compiler** [comment]
Hmm. The PR looks good (as far as I understand it :laughing:), but I'm wondering if it wouldn't be better to contribute this to [PyVirtualDisplay](https://github.com/ponty/PyVirtualDisplay.git) and then finally take care of #4.

**Comment by Kingdread** [comment]
I'll see if I can open a PR for PyVirtualDisplay later, and if they accept it, I'll work on #4 :wink:

**Comment by The-Compiler** [comment]
Awesome, thank you!

**Comment by Kingdread** [comment]
I've opened ponty/PyVirtualDisplay#17 for now, let's see what happens with the patch :wink:

**Comment by The-Compiler** [comment]
Just released v1.0.0 with that in :wink:
